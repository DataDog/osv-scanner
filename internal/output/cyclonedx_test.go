package output_test

import (
	"bytes"
	"testing"

	"github.com/datadog/osv-scanner/internal/output"
	"github.com/datadog/osv-scanner/internal/testutility"
	"github.com/datadog/osv-scanner/pkg/models"
)

func TestPrintCycloneDX14Results_WithDependencies(t *testing.T) {
	t.Parallel()

	testOutputWithArtifacts(t, func(t *testing.T, args outputTestCaseArgs) {
		t.Helper()

		outputWriter := &bytes.Buffer{}
		err := output.PrintCycloneDXResults(args.vulnResult, models.CycloneDXVersion14, outputWriter)

		if err != nil {
			t.Errorf("%v", err)
		}

		testutility.NewSnapshot().MatchText(t, outputWriter.String())
	})
}

func TestPrintCycloneDX15Results_WithDependencies(t *testing.T) {
	t.Parallel()

	testOutputWithArtifacts(t, func(t *testing.T, args outputTestCaseArgs) {
		t.Helper()

		outputWriter := &bytes.Buffer{}
		err := output.PrintCycloneDXResults(args.vulnResult, models.CycloneDXVersion15, outputWriter)

		if err != nil {
			t.Errorf("%v", err)
		}

		testutility.NewSnapshot().MatchText(t, outputWriter.String())
	})
}
