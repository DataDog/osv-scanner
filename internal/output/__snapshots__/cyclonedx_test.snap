
[TestPrintCycloneDX14Results_WithDependencies/multiple_sources_with_no_packages - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1
}

---

[TestPrintCycloneDX14Results_WithDependencies/no_sources - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1
}

---

[TestPrintCycloneDX14Results_WithDependencies/one_source_with_multiple_artifacts,_used_by_one_of_them - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:maven/dev.foo/lib1@1.0-SNAPSHOT",
      "type": "library",
      "name": "dev.foo:lib1",
      "version": "1.0-SNAPSHOT",
      "purl": "pkg:maven/dev.foo/lib1@1.0-SNAPSHOT"
    },
    {
      "bom-ref": "pkg:maven/dev.foo/lib2@1.0-SNAPSHOT",
      "type": "library",
      "name": "dev.foo:lib2",
      "version": "1.0-SNAPSHOT",
      "purl": "pkg:maven/dev.foo/lib2@1.0-SNAPSHOT"
    }
  ],
  "dependencies": [
    {
      "ref": "pom.xml",
      "dependsOn": [
        "lib1/pom.xml",
        "lib2/pom.xml"
      ]
    }
  ]
}

---

[TestPrintCycloneDX14Results_WithDependencies/one_source_with_no_packages - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1
}

---

[TestPrintCycloneDX14Results_WithDependencies/one_source_with_one_artifact,_no_dependsOn - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:maven/com.nobody/mine1@1.2.3",
      "type": "library",
      "name": "com.nobody:mine1",
      "version": "1.2.3",
      "purl": "pkg:maven/com.nobody/mine1@1.2.3"
    }
  ]
}

---

[TestPrintCycloneDX14Results_WithDependencies/one_source_with_one_artifact,_one_dependsOn - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:maven/com.nobody/mine1@1.2.3",
      "type": "library",
      "name": "com.nobody:mine1",
      "version": "1.2.3",
      "purl": "pkg:maven/com.nobody/mine1@1.2.3"
    }
  ]
}

---

[TestPrintCycloneDX14Results_WithDependencies/one_source_with_one_package,_no_artifacts - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:npm/com.nobody%3Amine1@1.2.3",
      "type": "library",
      "name": "com.nobody:mine1",
      "version": "1.2.3",
      "purl": "pkg:npm/com.nobody%3Amine1@1.2.3"
    }
  ]
}

---

[TestPrintCycloneDX15Results_WithDependencies/multiple_sources_with_no_packages - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1
}

---

[TestPrintCycloneDX15Results_WithDependencies/no_sources - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1
}

---

[TestPrintCycloneDX15Results_WithDependencies/one_source_with_multiple_artifacts,_used_by_one_of_them - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:maven/dev.foo/lib1@1.0-SNAPSHOT",
      "type": "library",
      "name": "dev.foo:lib1",
      "version": "1.0-SNAPSHOT",
      "purl": "pkg:maven/dev.foo/lib1@1.0-SNAPSHOT"
    },
    {
      "bom-ref": "pkg:maven/dev.foo/lib2@1.0-SNAPSHOT",
      "type": "library",
      "name": "dev.foo:lib2",
      "version": "1.0-SNAPSHOT",
      "purl": "pkg:maven/dev.foo/lib2@1.0-SNAPSHOT"
    }
  ],
  "dependencies": [
    {
      "ref": "pom.xml",
      "dependsOn": [
        "lib1/pom.xml",
        "lib2/pom.xml"
      ]
    }
  ]
}

---

[TestPrintCycloneDX15Results_WithDependencies/one_source_with_no_packages - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1
}

---

[TestPrintCycloneDX15Results_WithDependencies/one_source_with_one_artifact,_no_dependsOn - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:maven/com.nobody/mine1@1.2.3",
      "type": "library",
      "name": "com.nobody:mine1",
      "version": "1.2.3",
      "purl": "pkg:maven/com.nobody/mine1@1.2.3"
    }
  ]
}

---

[TestPrintCycloneDX15Results_WithDependencies/one_source_with_one_artifact,_one_dependsOn - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:maven/com.nobody/mine1@1.2.3",
      "type": "library",
      "name": "com.nobody:mine1",
      "version": "1.2.3",
      "purl": "pkg:maven/com.nobody/mine1@1.2.3"
    }
  ]
}

---

[TestPrintCycloneDX15Results_WithDependencies/one_source_with_one_package,_no_artifacts - 1]
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1,
  "components": [
    {
      "bom-ref": "pkg:npm/com.nobody%3Amine1@1.2.3",
      "type": "library",
      "name": "com.nobody:mine1",
      "version": "1.2.3",
      "purl": "pkg:npm/com.nobody%3Amine1@1.2.3"
    }
  ]
}

---
